manifests/server/grant.pp

```diff
@@ -337,8 +407,13 @@
     require          => Class['postgresql::server']
   }
 
-  if($role != undef and defined(Postgresql::Server::Role[$role])) {
-    Postgresql::Server::Role[$role]->Postgresql_psql["${title}: grant:${name}"]
+  if($role != undef) {
+    if ($_lowercase_role =~ /^group (.*)/ and defined(Postgresql::Server::Dbgroup["#$1"])) {
+      Postgresql::Server::Dbgroup["#$1"]->Postgresql_psql["${title}: grant:${name}"]
+    }
+    if defined(Postgresql::Server::Role[$role]) {
```

Comment:
why the `if defined`? shouldn't the role already exist? If it doesn't this should be an error, right?

Reply:
This is a holdover from the original module. I would assume this is to allow the role to be defined independently of the grant (ie, in later reviews), and to specify an ordering when (and only when) it exists (since puppet will topologically sorts class evaluation instead of compiling each class in evaluation order).

For that reason, this seems fine to me.

Reply:
This is a holdover from the original module. I would assume this is to allow the role to be defined independently of the grant (ie, in later reviews), and to specify an ordering when that exists.